core/commands/reflog.py

```diff
@@ -0,0 +1,37 @@
+import sublime
+from ..ui_mixins.quick_panel import show_paginated_panel
+from ...common import util
+
+
+class RefLogMixin(object):
+
+    _limit = 6000
+
+    def run(self):
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        show_paginated_panel(self.log_generator(), self.do_action, limit=self._limit)
+
+    def log_generator(self):
+        skip = 0
+        while True:
+            logs = self.reflog(limit=self._limit, skip=skip)
+            if not logs:
+                break
+            for l in logs:
+                yield ([l.reflog_selector + " " + l.reflog_name,
```

Comment:
I would prefer we used string formatting here; [string concat is inefficient][1] and, arguably, harder to read.

[1]: https://wiki.python.org/moin/PythonSpeed/PerformanceTips#String_Concatenation